resources/design-document.md

```diff
@@ -0,0 +1,267 @@
+
+# 5. Proposed Architecture Overview
+
+## Project Phase: Minimum Lovable Product (MLP) Delivery
+### Description:
+In this initial phase, our goal is to deliver the Minimum Lovable Product (MLP) for our software project. The MLP will encompass core functionalities, including project creation, retrieval, and updating. Additionally, it will involve fetching content from external APIs to populate a word pool and workspace, allowing customization to meet specific user needs.
+
+### Key Components:
+
+#### API Gateway:
+
+We will utilize AWS API Gateway to manage and expose a set of RESTful endpoints to the outside world. These endpoints are designed to handle project-related actions.
+
+#### AWS Lambda:
+
+AWS Lambda functions will be used to implement serverless compute for our project. The Lambda functions we will create will each be responsible for specific actions related to project management and content modification.
+
+###### Endpoints:
+
+The endpoints defined in section 6.2 are designed to facilitate various project-related operations. These endpoints include, but are not limited to:
+
+* Project Creation: An endpoint to create new projects.
+* Project Deletion: An endpoint to delete a project.
+* Project Retrieval: Endpoints for retrieving project details and content.
+* Project Update: Endpoints for updating existing projects.
+* Word Pool Population: Endpoints for fetching content from external APIs to populate a word pool.
+* Workspace Customization: Endpoints for customizing the workspace within a project.
+
+###### Functionality:
+
+* Project Management:
+
+1. Users can create new projects, delete existing projects and retrieve project information.
+2. Projects can be updated as needed to accommodate changes or additions.
+
+* Content Integration:
+
+1. External APIs will be accessed to fetch content, which is then populated into the word pool and workspace.
+
+###### Scalability:
+
+* The serverless architecture offered by AWS Lambda allows for easy scalability. Additional Lambda functions can be added to handle increased load as the project grows.
+
+###### Security:
+
+* Security and authentication will be managed through AWS Cognito to protect user data and project content.
+
+###### Customization:
+
+* Users will have the ability to customize and modify the content within their projects and word pools to suit their specific requirements.
+
+
+# 6. API
+
+## 6.1. Public Models
+
+```
+// ProjectModel
+
+String userId;
+String projectId;
+String projectName;
+Date creationDate;
+Date lastModified;
+List<String> wordPool;
+List<String> workspace;
+```
+```
+// WordpoolModel
+
+String userId;
+String wordPoolId;
+String wordPoolName;
+Date creationDate;
+Date lastModified;
+List<String> wordPool;
+
+```
+
+
+## 6.2. Endpoints
+
+### Project
+
+* Accepts POST requests to /project
+###### Accepts a projectName and userId and returns the corresponding ProjectModel including a unique projectId assigned by the Project Service.
```

Comment:
If userId is going to be the cognito id, you can get that directly from cognito, instead of it being part of the request body.